google/resource_bigquery_dataset.go

```diff
@@ -274,12 +274,32 @@ func resourceBigQueryDatasetDelete(d *schema.ResourceData, meta interface{}) err
 
 	log.Printf("[INFO] Deleting BigQuery dataset: %s", d.Id())
 
-	projectID, datasetID := resourceBigQueryDatasetParseID(d.Id())
+	id, err := parseBigQueryDatasetId(d.Id())
+	if err != nil {
+		return err
+	}
 
-	if err := config.clientBigQuery.Datasets.Delete(projectID, datasetID).Do(); err != nil {
+	if err := config.clientBigQuery.Datasets.Delete(id.Project, id.DatasetId).Do(); err != nil {
 		return err
 	}
 
 	d.SetId("")
 	return nil
 }
+
+type bigQueryDatasetId struct {
+	Project, DatasetId string
+}
+
+func parseBigQueryDatasetId(id string) (*bigQueryDatasetId, error) {
+	parts := strings.Split(id, ":")
+
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("Invalid BigQuery dataset specifier. Expecting {project}:{dataset-id}, got %s", id)
```

Comment:
Some project IDs also have colons, unfortunately - back in the old days they namespaced projects with "example.com:project-name".  Maybe do a regex match - the project resource has a project regex, and there should be an easy one for dataset ID also.

Reply:
Super good point! Done.